server/models.py

```diff
@@ -1639,3 +1639,71 @@ class CanvasAssignment(Model):
     @property
     def url(self):
         return '{}/assignments/{}'.format(self.canvas_course.url, self.external_id)
+
+#########
+# Files #
+#########
+
+class ExternalFile(Model):
+
+    id = db.Column(db.Integer, primary_key=True)
+    # Bucket/Folder that the file is stored in
+    container = db.Column(db.String(255), nullable=False)
+    filename = db.Column(db.String(1024), nullable=False)
+    object_name = db.Column(db.String(1024), nullable=False)
+
+    is_staff = db.Column(db.Boolean, nullable=False)
+    deleted = db.Column(db.Boolean, default=False)
+
+    course_id = db.Column(
+        db.Integer, db.ForeignKey('course.id'), index=True, nullable=False
+    )
+    course = db.relationship('Course')
+
+    user_id = db.Column(
+        db.Integer, db.ForeignKey('user.id'), index=True, nullable=False
+    )
+    user = db.relationship('User')
+
+    assignment_id = db.Column(
+        db.Integer, db.ForeignKey('assignment.id'), index=True
+    )
+    assignment = db.relationship('Assignment')
+
+    @property
+    def object(self):
+        with storage.use(self.container) as container:
+            return container.get(self.object_name)
+
+    def delete(self):
```

Comment:
It's possible that we'll have a file dashboard and a way to delete files so I wanted to provide support now (to avoid another migration)